crates/router/src/connector/applepay/transformers.rs

```diff
@@ -55,48 +87,133 @@ impl TryFrom<&types::PaymentsSessionRouterData> for ApplepaySessionRequest {
             .get_required_value("connector_meta_data")
             .change_context(errors::ConnectorError::NoConnectorMetaData)?;
 
-        let session_object: SessionObject = metadata
-            .parse_value("SessionObject")
+        let metadata: ApplePayMetaData = metadata
+            .parse_value("ApplePayMetaData")
             .change_context(errors::ConnectorError::RequestEncodingFailed)?;
 
         Ok(Self {
-            merchant_identifier: session_object.merchant_identifier,
-            display_name: session_object.display_name,
-            initiative: session_object.initiative,
-            initiative_context: session_object.initiative_context,
+            merchant_identifier: metadata.session_object.merchant_identifier,
+            display_name: metadata.session_object.display_name,
+            initiative: metadata.session_object.initiative,
+            initiative_context: metadata.session_object.initiative_context,
         })
     }
 }
 
-impl<F, T>
-    TryFrom<types::ResponseRouterData<F, ApplepaySessionResponse, T, types::PaymentsResponseData>>
-    for types::RouterData<F, T, types::PaymentsResponseData>
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            ApplepaySessionResponse,
+            types::PaymentsSessionData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsSessionData, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: types::ResponseRouterData<F, ApplepaySessionResponse, T, types::PaymentsResponseData>,
+        item: types::ResponseRouterData<
+            F,
+            ApplepaySessionResponse,
+            types::PaymentsSessionData,
+            types::PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
+        let metadata = item
+            .data
+            .connector_meta_data
+            .to_owned()
+            .get_required_value("connector_meta_data")
+            .change_context(errors::ConnectorError::NoConnectorMetaData)?;
+
+        let metadata: ApplePayMetaData = metadata
+            .parse_value("ApplePayMetaData")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+
+        let amount_info = AmountInfo {
+            label: "Demo".to_string(),
```

Comment:
Why these where hard coded?

Reply:
label type is always final, label hardcoding is removed as it appears on their bank or credit card statement.